plaso/engine/artifact_filters.py

```diff
@@ -0,0 +1,259 @@
+      find_specs dict[artifacts.artifact_types]:  Dictionary containing
+          find_specs.
+    """
+    for glob_path in self._ExpandRecursiveGlobs(path_entry):
+      for path in path_helper.PathHelper.ExpandUserHomeDirPath(
+          glob_path, self._knowledge_base.user_accounts):
+        if '%' in path:
+          path = path_helper.PathHelper.ExpandWindowsPath(
+              path, environment_variables)
+
+        if not path.startswith('/') and not path.startswith('\\'):
+          logging.warning((
+              'The path filter must be defined as an absolute path: '
+              '{0:s}').format(path))
+          continue
+
+        # Convert the path filters into a list of path segments and
+        # strip the root path segment.
+        path_segments = path.split(separator)
+
+        # If the source didn't specify a separator, '/' is returned by
+        # default from ForensicArtifacts, this is sometimes wrong.  Thus,
+        # need to check if '\' characters are still present and split on those.
+        if len(path_segments) == 1 and '\\' in path_segments[0]:
+          logging.warning('Potentially bad separator = {0:s} , trying \'\\\''
+                          .format(separator))
+          path_segments = path.split('\\')
+        path_segments.pop(0)
+
+        if not path_segments[-1]:
+          logging.warning(
+              'Empty last path segment in path filter: {0:s}'.format(path))
+          path_segments.pop(-1)
+
+        try:
+          find_spec = file_system_searcher.FindSpec(
+              location_glob=path_segments, case_sensitive=False)
+        except ValueError as exception:
+          logging.error((
+              'Unable to build find spec for path: {0:s} with error: {1!s}'
+          ).format(path, exception))
+          continue
+        if artifact_types.TYPE_INDICATOR_FILE not in find_specs:
+          find_specs[artifact_types.TYPE_INDICATOR_FILE] = []
+        find_specs[artifact_types.TYPE_INDICATOR_FILE].append(
+            find_spec)
+
+  def BuildFindSpecsFromRegistryArtifact(
+      self, key_entry, find_specs):
+    """Build find specifications from a Windows registry artifact type.
+
+    Args:
+      key_entry (str):  Current file system key to add.
+      find_specs dict[artifacts.artifact_types]:  Dictionary containing
+          find_specs.
+    """
+    for key in self._ExpandRecursiveGlobs(key_entry):
+      if '%%' in key:
+        logging.error((
+            'Unable to expand path filter: {0:s}').format(key))
+        continue
+      find_spec = registry_searcher.FindSpec(key_path_glob=key)
+      if artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY not in find_specs:
+        find_specs[artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY] = []
+
+      find_specs[artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY].append(
+          find_spec)
+
+  def _CheckKeyCompatibility(self, key):
+    """Check if a registry key is compatible with dfwinreg.
```

Comment:
Check => Checks
registry => Registry (since its a name in the context of Windows Registry), preferably use "Windows Registry" (seeing the term registry is used in different contexts)

Reply:
Done.